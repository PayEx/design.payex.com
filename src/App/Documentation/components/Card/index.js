import React, { Component } from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, Property, DocToc, ExperimentalComponentAlert } from "#";
import CardComponent from "@/Card";
import MediaObject from "@/MediaObject";

const DefaultBehaviour = () => (
    <>
        <h2 id="default-behaviour">Default behaviour</h2>
        <p>
            Wrap indiviual cards in the <Property value=".cards" /> class to display them lined up next to each other.
            When the window reaches <Link to="/docs/core/breakpoints">breakpoint</Link> sm the cards will switch to mobile view and take up one row each and stretch to fill the width of the container.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="cards">
                <CardComponent
                    type="default"
                    title="Card default"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@payex.com" textSmall muted />
                </CardComponent>

                <CardComponent
                    type="default"
                    title="Card default"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="account_balance" heading="725 NOK" text="Outstanding balance" textSmall muted />
                </CardComponent>

                <CardComponent
                    type="default"
                    title="Card default"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                </CardComponent>
            </div>
        </ComponentPreview>
    </>
);

const CardTypes = () => (
    <>
        <h2 id="card-types">Card Types</h2>
        <p>To decide how your cards will look use <Property value=".card-default" />, <Property value=".card-brand" />, or <Property value=".card-light" /></p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="cards">
                <CardComponent
                    type="default"
                    title="Card default"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@payex.com" textSmall muted />
                </CardComponent>

                <CardComponent
                    type="brand"
                    title="Card brand"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="account_balance" heading="725 NOK" text="Outstanding balance" textSmall muted />
                </CardComponent>

                <CardComponent
                    type="light"
                    title="Card light"
                    imgSrc="https://picsum.photos/300"
                    text="This is the cards text"
                    smallText="And the small text"
                    btnTxt="Button"
                    footerTxt="This is some text that goes in the footer."
                    footerLinkTxt="You wanna click this link?"
                >
                    <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                </CardComponent>
            </div>
        </ComponentPreview>
    </>
);

const CardText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <ExperimentalComponentAlert />
        <p className="lead">Card...</p>
        <DefaultBehaviour />
        <CardTypes />
    </div>
);

class Card extends Component {
    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <CardText />
                    <DocToc component={CardText} />
                </div>
            </div>
        );
    }
}

export default Card;

/* For testing */
export { DefaultBehaviour, CardTypes, CardText };
