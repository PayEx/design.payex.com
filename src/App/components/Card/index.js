import React from "react";
import PropTypes from "prop-types";

const Card = ({ type, title, imgSrc, text, smallText, btnTxt, footerTxt, footerLink, footerLinkTxt, children }) => (
    <div className={`card${type ? ` card-${type}` : ""}`}>
        {title ? <header>
            <h3>
                <a href="#">{title}</a>
            </h3>
        </header> : null}
        {imgSrc ? <div className="card-img">{"\n"}
            <img src={imgSrc} alt=""/>{"\n"}
        </div> : null}
        <div className="highlight">
            {children}
        </div>
        <div className="card-body">
            {text ? <p>{text}</p> : null}{"\n"}
            {smallText ? <small>{smallText}</small> : null}
        </div>
        <footer>
            {btnTxt ? <button type="button" className="btn btn-primary btn-block">{btnTxt}</button> : null}
            {footerTxt ? <div className="footer-text"><p>{footerTxt}</p></div> : null}
            {footerLinkTxt ? <div className="footer-link"><a href={footerLink || "#"}>{footerLinkTxt}</a></div> : null}
        </footer>
    </div>
);

Card.propTypes = {
    type: PropTypes.oneOf(["brand", "default", "light"]),
    title: PropTypes.string,
    imgSrc: PropTypes.string,
    icon: PropTypes.string,
    text: PropTypes.string,
    smallText: PropTypes.string,
    btnTxt: PropTypes.string,
    footerTxt: PropTypes.string,
    footerLink: PropTypes.string,
    footerLinkTxt: PropTypes.string
};

export default Card;
