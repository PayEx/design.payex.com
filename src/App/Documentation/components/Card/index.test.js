import React from "react";
import { shallow } from "enzyme";

import Card, { DefaultBehaviour, CardTypes, CardText } from "./index";

describe("Documentation: Card", () => {
    it("is defined", () => {
        expect(Card).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Card />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("DefaultBehaviour", () => {
        it("is defined", () => {
            expect(DefaultBehaviour).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<DefaultBehaviour />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CardTypes", () => {
        it("is defined", () => {
            expect(CardTypes).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CardTypes />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CardText", () => {
        it("is defined", () => {
            expect(CardText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CardText />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
