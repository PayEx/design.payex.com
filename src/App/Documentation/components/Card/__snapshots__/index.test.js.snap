// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Card CardText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <ExperimentalComponentAlert />
  <p
    className="lead"
  >
    Card...
  </p>
  <DefaultBehaviour />
  <CardTypes />
</div>
`;

exports[`Documentation: Card CardTypes renders 1`] = `
<Fragment>
  <h2
    id="card-types"
  >
    Card Types
  </h2>
  <p>
    To decide how your cards will look use 
    <Property
      value=".card-default"
    />
    , 
    <Property
      value=".card-brand"
    />
    , or 
    <Property
      value=".card-light"
    />
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="cards"
    >
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card default"
        type="default"
      >
        <MediaObject
          heading="Bob Corlsan"
          icon="account_circle"
          muted={true}
          size="sm"
          text="bob.corlsan@payex.com"
          textSmall={true}
        />
      </Card>
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card brand"
        type="brand"
      >
        <MediaObject
          heading="725 NOK"
          icon="account_balance"
          muted={true}
          size="sm"
          text="Outstanding balance"
          textSmall={true}
        />
      </Card>
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card light"
        type="light"
      >
        <MediaObject
          heading="75%"
          icon="verified_user"
          muted={true}
          size="sm"
          text="Complete"
          textSmall={true}
        />
      </Card>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Card DefaultBehaviour renders 1`] = `
<Fragment>
  <h2
    id="default-behaviour"
  >
    Default behaviour
  </h2>
  <p>
    Wrap indiviual cards in the 
    <Property
      value=".cards"
    />
     class to display them lined up next to each other. When the window reaches 
    <Link
      replace={false}
      to="/docs/core/breakpoints"
    >
      breakpoint
    </Link>
     sm the cards will switch to mobile view and take up one row each and stretch to fill the width of the container.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="cards"
    >
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card default"
        type="default"
      >
        <MediaObject
          heading="Bob Corlsan"
          icon="account_circle"
          muted={true}
          size="sm"
          text="bob.corlsan@payex.com"
          textSmall={true}
        />
      </Card>
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card default"
        type="default"
      >
        <MediaObject
          heading="725 NOK"
          icon="account_balance"
          muted={true}
          size="sm"
          text="Outstanding balance"
          textSmall={true}
        />
      </Card>
      <Card
        btnTxt="Button"
        footerLinkTxt="You wanna click this link?"
        footerTxt="This is some text that goes in the footer."
        imgSrc="https://picsum.photos/300"
        smallText="And the small text"
        text="This is the cards text"
        title="Card default"
        type="default"
      >
        <MediaObject
          heading="75%"
          icon="verified_user"
          muted={true}
          size="sm"
          text="Complete"
          textSmall={true}
        />
      </Card>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Card renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <CardText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
